src/silo/test/has_mutation.test.cpp

```diff
@@ -0,0 +1,112 @@
+
+namespace {
+using silo::ReferenceGenomes;
+using silo::config::DatabaseConfig;
+using silo::config::ValueType;
+using silo::test::QueryTestData;
+using silo::test::QueryTestScenario;
+
+using boost::uuids::random_generator;
+
+nlohmann::json createDataWithSequences(
+   const std::string& nucleotideSequence,
+   const std::string& aminoAcidSequence
+) {
+   random_generator generator;
+   const auto primary_key = generator();
+
+   return {
+      {"metadata", {{"primaryKey", "id_" + to_string(primary_key)}}},
+      {"alignedNucleotideSequences", {{"segment1", nucleotideSequence}}},
+      {"unalignedNucleotideSequences", {{"segment1", {}}}},
+      {"alignedAminoAcidSequences", {{"gene1", aminoAcidSequence}}},
+      {"nucleotideInsertions", {{"segment1", {}}}},
+      {"aminoAcidInsertions", {{"gene1", {}}}}
+   };
+}
+
+const nlohmann::json DATA_SAME_AS_REFERENCE = createDataWithSequences("ATGCN", "M*");
+const nlohmann::json DATA_SAME_AS_REFERENCE2 = createDataWithSequences("ATGCN", "C*");
+const nlohmann::json DATA_WITH_ALL_N = createDataWithSequences("NNNNN", "M*");
+const nlohmann::json DATA_WITH_ALL_MUTATED = createDataWithSequences("CATTT", "X*");
+
+const auto DATABASE_CONFIG =
+   R"(
+defaultNucleotideSequence: "segment1"
+schema:
+  instanceName: "dummy name"
+  metadata:
+    - name: "primaryKey"
+      type: "string"
+  primaryKey: "primaryKey"
+)";
+
+const auto REFERENCE_GENOMES = ReferenceGenomes{
+   {{"segment1", "ATGCN"}},
+   {{"gene1", "M*"}},
+};
+
+const QueryTestData TEST_DATA{
+   .ndjson_input_data =
+      {DATA_SAME_AS_REFERENCE, DATA_SAME_AS_REFERENCE2, DATA_WITH_ALL_N, DATA_WITH_ALL_MUTATED},
+   .database_config = DATABASE_CONFIG,
+   .reference_genomes = REFERENCE_GENOMES
+};
+
+nlohmann::json createHasNucleotideMutationQuery(int position) {
+   return {
+      {"action", {{"type", "Aggregated"}}},
+      {"filterExpression",
+       {{"type", "HasNucleotideMutation"}, {"position", position}, {"sequenceName", "segment1"}}}
+   };
+}
+
+nlohmann::json createHasAminoAcidMutationQuery(int position) {
+   return {
+      {"action", {{"type", "Aggregated"}}},
+      {"filterExpression",
+       {{"type", "HasAminoAcidMutation"}, {"position", position}, {"sequenceName", "gene1"}}}
+   };
+}
+
+const QueryTestScenario HAS_NUCLEOTIDE_MUTATION = {
+   .name = "HAS_NUCLEOTIDE_MUTATION",
+   .query = createHasNucleotideMutationQuery(1),
+   .expected_query_result = nlohmann::json::parse(R"([{"count": 1}])")
+};
+
+const QueryTestScenario HAS_AMINO_ACID_MUTATION = {
+   .name = "HAS_AMINO_ACID_MUTATION",
+   .query = createHasAminoAcidMutationQuery(1),
+   .expected_query_result = nlohmann::json::parse(R"([{"count": 1}])")
+};
+
+const QueryTestScenario HAS_NUCLEOTIDE_MUTATION_OUT_OF_RANGE = {
+   .name = "HAS_NUCLEOTIDE_MUTATION_OUT_OF_RANGE",
+   .query = createHasNucleotideMutationQuery(2000),
+   .expected_error_message = "HasNucleotideMutation position is out of bounds 2000 > 5"
```

Comment:
I think we should also add the edges as tests: `position: 5` and `position: 6`. This has potential for an off-by-one error.